### DISCUSSION TITLE
[WASM] Custom login form no longer working after migration to .NET7

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#10462] [Q&A] [answered]
Hi,

I have a custom logon form (basically an OAuth2 workflow through Auth0 + AWS) which I called like
``` _ = Uno.Foundation.WebAssemblyRuntime.InvokeAsync($"(async () => {{ window.open('<URL-to-login-form>', '_self') }})();");```
... and which worked on UNO WASM before

However, with UNO WASM + .NET7 I now get these errors in browser console (screenshot).

Any idea what might be causing that?

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/26179242/202213827-9760ef02-2090-49c2-98ce-ad87ae0922e5.png">

**Comment by jeromelaban** [comment]
You seem to be having issues with the way pages are served (returning 404, most likely). Take a look at your network tab responses for more details.

**Reply by DierkDroth** [reply]
Thanks @jeromelaban 

I tried to isolate the cause of trouble and cut down the form - which I thought might be the cause of trouble - to a simple dummy HTML (xxx.html, attached) and I'm still getting this error.

Next I looked at the browser's Network tab but could not spot anything weird (screenshot below). What is it I should be looking for in particular?

[xxx.zip](https://github.com/unoplatform/uno/files/10024133/xxx.zip)

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/26179242/202248323-751abfb3-f327-46cc-adeb-ec869eef5d1b.png">

**Reply by jeromelaban** [reply]
By the looks of the contents, the files returned are all 1.6KB, which is not correct. Make sure that the content is located at the right place, using the right URI. You can open click on individual items to see the content in the network tab.

**Reply by DierkDroth** [reply]
@jeromelaban I'm sorry, I have no idea what you're talking about.

The files which have a size of 1.6KB are files which are 'managed' by UNO. AFAIK I'm not doing anthing with these files - at least that I would know. Hence I would not even know how to 'control' them by any means...

As I simply clicked through these files - without understanding what I'm doing here - I would see that their "Reponse" tabs show the content of the index.html file (including the `$(ADDITIONAL_CSS) $(ADDITIONAL_HEAD)` replacements) and their "Preview" tabs would show `This application requires Javascript and WebAssembly to be enabled.`

I completely agree this is weird. However, I'm a complete dummy here and would not know what to expect nor how to rectify the issue.

**Reply by jeromelaban** [reply]
We cannot troubleshoot without a repro, so if it works with a sample app, try recreating the steps until it breaks.

**Reply by DierkDroth** [reply]
@jeromelaban I got one step farther

FYI for anyone running into that problem.
Before I had something like this (URL was pointing to my OAuth0 login form):
```_ = Uno.Foundation.WebAssemblyRuntime.InvokeAsync($"(async () => {{ window.open('platform.uno/', '_self') }})();");```
... which I had to change to 
```await Uno.Foundation.WebAssemblyRuntime.InvokeAsync($"(async () => {{ window.open('https://platform.uno/', '_self') }})();");```

Note:
- additional `await'
- additional `https'

@jeromelaban now I'm stuck as my authentication backend gives me the OAuth0 redirect with the code like
`http://localhost:59516/code?code=Ud54UmQZE_nlQ0WTvteVScbz_wv4_614nSu0Yq-4_y8zu`
... and the server throws error below (screenshot)

Would you have any idea where to look?

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/26179242/202368044-b5003fe7-6b94-403c-9d3a-f1d6c565346f.png">

**Comment by MasterThreadInf** [comment]
Do you have sample solution for it ? 
Are you using https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Tutorials/Authentication/HowTo-Authentication.html ?

**Reply by DierkDroth** [reply]
No, I'm not using any package at all. This is all custom code and 'naked' .NET.

**Reply by MasterThreadInf** [reply]
Ok

**Comment by DierkDroth** [comment, marked as answer]
I was able to resolve the adding the offending port 59516 to the launchSettings.json file:
` "applicationUrl": "http://localhost:5000;http://localhost:59516",`
